src/AppShed/Extensions/SpreadsheetBundle/Service/GeoService.php

```diff
@@ -11,33 +11,46 @@
  */
 class GeoService {
 
+    private $geoClient;
+
+    public function __construct(Client $geoClient)
+    {
+        $this->geoClient = $geoClient;
+    }
+
     public function getGeo($address)
     {
-        $guzzle = new Client();
 
-        $address = urlencode($address);
+        try {
+            $geoResponse = $this->geoClient->get(
+                'geocode/json',
+                [
+                    'query' => [
+                        'address' => $address
+                    ]
+                ]
+            );
 
-        $url = "http://maps.google.com/maps/api/geocode/json?sensor=false&address={$address}";
+            if($geoResponse->getStatusCode()==200){
```

Comment:
A nice way to do this, is to write some like - the point is to exit early, instead of having `if { if { if {` and more and more spaces, makes the code more readable.

    if ($geoResponse->getStatusCode() != 200) {
        return false;
    }

    $data = $res->json();

    if ($resp['status'] != 'OK' || !count($resp['results'])) {
        return false;
    }